port/port.js

```diff
@@ -0,0 +1,143 @@
+import { getUniqueId } from './unique-id.js';
+import { validateEvent } from './validate-event.js';
+
+export class Port {
+
+	constructor(endPoint, targetOrigin, options) {
+		options = options || {};
+
+		this._connectQueue = [];
+		this._pluginStartupValues = [];
+		this._debugEnabled = options.debug || false;
+		this._endPoint = endPoint;
+		this._eventHandlers = {};
+		this._isConnected = false;
+		this._isOpen = false;
+		this._messageHandlers = {};
+		this._onCloseCallbacks = [];
+		this._targetOrigin = targetOrigin;
+
+		this._id = getUniqueId();
```

Comment:
While the `uuid` package does have an ESM wrapper, it would need to be transpiled to use directly in the browser (e.g. in a web component). My understanding is that we need this to be a unique ID, but don't necessarily need it to be random, so I've scammed the functionality from here and changed the namespace to avoid any indirect coupling between core and ifrau: https://github.com/BrightspaceUI/core/blob/main/helpers/uniqueId.js